rflx/model/model.py

```diff
@@ -81,13 +81,15 @@ def __check_duplicates(self) -> RecordFluxError:
                     Severity.ERROR,
                     s.location,
                 )
+                if s.identifier in types:
+                    prev_loc = [d for d in types if d == s.identifier][0].location
```

Comment:
~This could be written as `prev_loc = types[types.index(s.identifier)].location`.~
Thinking a bit more about this, in which way is this different from `prev_loc = s.identifier.location`? The expression `[d for d in types if d == s.identifier][0]` always returns `s.identifier`. So unless there is a difference between two elements that compare as equal this expression seems useless to me. If there is a difference there should be a comment (and in this case my crossed out note is still valid).

Reply:
When comparing two objects of the `ID` class the location is ignored. `types` is a dict which maps the `ID` of a type to the actual type declaration. What we're interested in at this point is the ID (i.e. the key) of the **original** type which has been put into `types` earlier (**not** the location of the respective element). [Python does not seem to have a way to retrieve the original key in this case](https://stackoverflow.com/questions/64955067/get-value-of-dict-key-without-iterating-items), hence we need to iterate. The code you propose retrieves the location of the element, not the key. I tried to explain this in 20c9f389.

Reply:
My original comment assumed that `types` was a list and not a dict so it's invalid anyway. I think with the explanation its much better to understand.